src/geopackage/sqlite/database.cpp

```diff
@@ -0,0 +1,124 @@
+#include "NGen_SQLite.hpp"
+
+using namespace geopackage;
+
+/**
+ * Get a runtime error based on a function and code.
+ *
+ * @param f String denoting the function where the error originated
+ * @param code sqlite3 result code
+ * @param extra additional messages to add to the end of the error
+ * @return std::runtime_error
+ */
+std::runtime_error sqlite_error(const std::string& f, int code, const std::string& extra = "")
+{
+    std::string errmsg = f + " returned code "
+                           + std::to_string(code)
+                           + " (msg: "
+                           + std::string(sqlite3_errstr(code))
+                           + ")";
+
+    if (!extra.empty()) {
+      errmsg += " ";
+      errmsg += extra;
+    }
+
+    return std::runtime_error(errmsg);
+}
+
+sqlite::sqlite(const std::string& path)
+{
+    sqlite3* conn;
+    int      code = sqlite3_open_v2(path.c_str(), &conn, SQLITE_OPEN_READONLY, NULL);
+    if (code != SQLITE_OK) {
+        throw sqlite_error("sqlite3_open_v2", code);
+    }
+    this->conn = sqlite_t(conn);
+}
+
+sqlite::sqlite(sqlite&& db)
+{
+    this->conn = std::move(db.conn);
+    this->stmt = db.stmt;
+}
+
+sqlite& sqlite::operator=(sqlite&& db)
+{
+    this->conn = std::move(db.conn);
+    this->stmt = db.stmt;
+    return *this;
+}
```

Comment:
Should these set the RHS `stmt` member to `nullptr`? Just saying `this->stmt = std::move(db.stmt);` would accomplish that, since `std::shared_ptr::{shared_ptr,operator=}(shared_ptr&&)` does that.

At that point, we're effectively just writing out `sqlite(sqlite&&) = default;` in long form, and should just do that.

Reply:
I think these definitions were a remnant of earlier iterations when there was a bit more complexity to the query/statement relationship. I agree that these should just be `= default`.